mempool/src/shared_mempool/mod.rs

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+pub mod network;
+mod runtime;
+/// Mempool types that are used by external crates
+pub mod types;
+pub use runtime::bootstrap;
+#[cfg(feature = "fuzzing")]
+pub(crate) use runtime::start_shared_mempool;
+mod coordinator;
+mod tasks;
+
+use self::network::MempoolNetworkSender;
+use crate::core_mempool::CoreMempool;
+use futures::{channel::mpsc::UnboundedSender, Stream};
+use libra_config::config::MempoolConfig;
+use libra_types::PeerId;
+use std::{
+    collections::HashMap,
+    pin::Pin,
+    sync::{Arc, Mutex},
+};
+use storage_client::StorageRead;
+use tokio::sync::RwLock;
+use vm_validator::vm_validator::TransactionValidation;
+
+/// Struct that owns all dependencies required by shared mempool routines
+#[derive(Clone)]
+pub(crate) struct SharedMempool<V>
```

Comment:
nit: I'd create separate types.rs for this. Specially things like IntervalStream do not fit top level mod.rs